### ISSUE TITLE
[Hyprland] Workspaces not appearing any longer

### ISSUE THREAD

**Issue by pescepalla** [MaxVerevkin/i3bar-river#20]
I've recently updated i3bar-river and whereas before it displayed workspaces, now they are not visible.

My config:

```
command = "i3status-rs"

background = "#00000000"

font = "JetBrainsMono Nerd Font 10"
height = 25
margin_top = 2
margin_bottom = 0
margin_left = 2
margin_right = 2
separator_width = 0
tags_r = 6
tags_padding = 10
blocks_r = 6
blocks_overlap = 5

position = "top"
hide_inactive_tags = true
invert_touchpad_scrolling = true
show_layout_name = true
blend = true
show_mode = true

[output.eDP-1]
enable = true
```

Using i3bar-river-git with Hyprland, Arch Linux.

**Comment by MaxVerevkin** [comment]
See #19

**Comment by pescepalla** [comment]
I see, thank you. Sad to have to drop the only non-bloated and easy-to-configure bar I had found. Will find an alternative.

**Comment by MaxVerevkin** [comment]
You can use an older version of i3bar-river if you stick with hyprland 29.0 (they removed the protocol in 29.1, a "minor" release...).

**Comment by MaxVerevkin** [comment]
> Sad to have to drop

Not if someone implements their custom IPC protocol :)

**Comment by pescepalla** [comment]
Unfortunately, I lack the technical skills, but thank you all the same. It was good till it lasted.

**Comment by MaxVerevkin** [comment]
I cannot guarantee anything but maybe I'll implement it myself. I'll see how it goes.

**Comment by MaxVerevkin** [comment]
From my limited testing this seems to work - 74cc46a. Hyprland-specific code fits in less than 200 LOC, while in waybar this numeber is > 700, so something is probably missing.

**Comment by pescepalla** [comment]
That was fast!

I've downloaded and built it. It still does not show any workspaces, unfortunately.

It also won't take the `blend` and `show_mode` options as well as the `output` overrides.
![2023-09-07-091038_hyprshot](https://github.com/MaxVerevkin/i3bar-river/assets/46329185/96d6c03c-7abe-46d0-b32a-da621a1dfa29)

**Comment by MaxVerevkin** [comment]
Are you sure you are building the latest master? Blending and output overrides were not affected.

**Comment by pescepalla** [comment]
I did a `git clone https://github.com/MaxVerevkin/i3bar-river.git` and built following the instructions.

This version does not even take options. If I pass `--version` or `--help`, it just launches. This is different from the behaviour of the AUR version.

![2023-09-07-130130_hyprshot](https://github.com/MaxVerevkin/i3bar-river/assets/46329185/1191e716-0fad-442c-8d64-f5efc8ad5016)

**Comment by MaxVerevkin** [comment]
> cargo install --locked i3bar-river

This installs the latest published version. Clone the repo, cd into it, run `cargo install --locked --path .`

**Comment by pescepalla** [comment]
Can confirm this one works! Many thanks.

**Comment by Henriquelay** [comment]
@MaxVerevkin Hyprland support is still suggested in the repo description. In fact, it bloated my searches a bit.

**Comment by MaxVerevkin** [comment]
> @MaxVerevkin Hyprland support is still suggested in the repo description. In fact, it bloated my searches a bit.

@Henriquelay In fact, Hyprland support is still implemented and is working.

**Comment by Henriquelay** [comment]
Is it officially supported then? Or it simply __works__?
I tried, it does work to the limited extent that I've tried, but that is not the same as supporting it. All the issues/commit messages suggest that it is in fact not supported, and is a "8t works, figure out the rest" deal.

I'm not complaining if it is the latter: this is amazing software either way (along with your other awesome project i3status-rs), this is just better cleared out

Mar 21, 2024 02:40:39 Max Verevkin ***@***.***>:
